cranelift/codegen/src/isa/arm64/mod.rs

```diff
@@ -1,132 +1,218 @@
 
-fn isa_constructor(
-    triple: Triple,
-    shared_flags: shared_settings::Flags,
-    builder: shared_settings::Builder,
-) -> Box<dyn TargetIsa> {
-    Box::new(Isa {
-        triple,
-        isa_flags: settings::Flags::new(&shared_flags, builder),
-        shared_flags,
-    })
+/// An ARM64 backend.
+pub struct Arm64Backend {
+    flags: settings::Flags,
 }
 
-impl TargetIsa for Isa {
-    fn name(&self) -> &'static str {
-        "arm64"
-    }
-
-    fn triple(&self) -> &Triple {
-        &self.triple
-    }
-
-    fn flags(&self) -> &shared_settings::Flags {
-        &self.shared_flags
-    }
-
-    fn register_info(&self) -> RegInfo {
-        registers::INFO.clone()
+impl Arm64Backend {
+    /// Create a new ARM64 backend with the given (shared) flags.
+    pub fn new_with_flags(flags: settings::Flags) -> Arm64Backend {
+        Arm64Backend { flags }
     }
 
-    fn encoding_info(&self) -> EncInfo {
-        enc_tables::INFO.clone()
+    fn compile_vcode(&self, mut func: Function, flags: &settings::Flags) -> VCode<inst::Inst> {
+        // This performs lowering to VCode, register-allocates the code, computes
+        // block layout and finalizes branches. The result is ready for binary emission.
+        let abi = Box::new(abi::ARM64ABIBody::new(&func));
+        compile::compile::<Arm64Backend>(&mut func, self, abi, flags)
     }
+}
 
-    fn legal_encodings<'a>(
-        &'a self,
-        func: &'a ir::Function,
-        inst: &'a ir::InstructionData,
-        ctrl_typevar: ir::Type,
-    ) -> Encodings<'a> {
-        lookup_enclist(
-            ctrl_typevar,
-            inst,
-            func,
-            &enc_tables::LEVEL1_A64[..],
-            &enc_tables::LEVEL2[..],
-            &enc_tables::ENCLISTS[..],
-            &enc_tables::LEGALIZE_ACTIONS[..],
-            &enc_tables::RECIPE_PREDICATES[..],
-            &enc_tables::INST_PREDICATES[..],
-            self.isa_flags.predicate_view(),
-        )
+impl MachBackend for Arm64Backend {
+    fn compile_function(
+        &self,
+        func: Function,
+        want_disasm: bool,
+    ) -> CodegenResult<MachCompileResult> {
+        let flags = self.flags();
+        let vcode = self.compile_vcode(func, flags);
+        let sections = vcode.emit();
+        let frame_size = vcode.frame_size();
+
+        let disasm = if want_disasm {
+            Some(vcode.show_rru(Some(&create_reg_universe())))
+        } else {
+            None
+        };
+
+        Ok(MachCompileResult {
+            sections,
+            frame_size,
+            disasm,
+        })
     }
 
-    fn legalize_signature(&self, sig: &mut Cow<ir::Signature>, current: bool) {
-        abi::legalize_signature(sig, &self.shared_flags, current)
+    fn name(&self) -> &'static str {
+        "arm64"
     }
 
-    fn regclass_for_abi_type(&self, ty: ir::Type) -> RegClass {
-        abi::regclass_for_abi_type(ty)
+    fn triple(&self) -> Triple {
+        FromStr::from_str("arm64").unwrap()
```

Comment:
This should be the triple passed to `isa_builder`. This value will result in a binary format of unknown, which means that `cranelift_object` will not work.

Reply:
Also `target-lexicon` calls it `aarch64`.

Reply:
I agree, we might as well standardize on the official name. Renamed the backend to `aarch64`. There are still a few vestigial remnants of `Arm64` around, e.g. the `Arm64Call` reloc type which is exposed in the public API.